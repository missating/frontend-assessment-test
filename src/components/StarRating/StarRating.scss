.starrating-container {
  display: flex;
  flex-wrap: wrap;

  &__star {
    cursor: pointer;
    width: 0.5em;
    height: 0.5em;
    background-color: var(--brand-color);
    -webkit-clip-path: polygon(
      50% 0%,
      63% 38%,
      100% 38%,
      69% 59%,
      82% 100%,
      50% 75%,
      18% 100%,
      31% 59%,
      0% 38%,
      37% 38%
    );
    clip-path: polygon(
      50% 0%,
      63% 38%,
      100% 38%,
      69% 59%,
      82% 100%,
      50% 75%,
      18% 100%,
      31% 59%,
      0% 38%,
      37% 38%
    );
  }
}
