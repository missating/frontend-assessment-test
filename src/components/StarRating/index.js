// thired party libraries
import React from 'react'
import PropTypes from 'prop-types';

// styles
import './StarRating.scss'

const starRating = ({ totalStars }) => (
  <div className="starrating-container">
    {[...Array(totalStars)].map((number, index) => (
      // eslint-disable-next-line react/no-array-index-key
      <div className="starrating-container__star" key={index} />
    ))}
  </div>
)

export default starRating;

starRating.propTypes = {
  totalStars: PropTypes.number.isRequired,
}
