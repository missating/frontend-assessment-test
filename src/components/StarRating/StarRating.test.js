// react library
// third-party libraries
import React from 'react';
import { shallow } from 'enzyme';

// components
import StarRating from './index'

describe('StarRating Component', () => {
  const initialProps = {
    totalStars: 3,
  }
  it('should be rendered properly', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = shallow(<StarRating {...initialProps} />);
    expect(wrapper.find('div').length).toEqual(4);
    expect(wrapper).toMatchSnapshot();
  });
})
