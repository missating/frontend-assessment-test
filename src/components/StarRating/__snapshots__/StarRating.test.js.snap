// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating Component should be rendered properly 1`] = `
<div
  className="starrating-container"
>
  <div
    className="starrating-container__star"
    key="0"
  />
  <div
    className="starrating-container__star"
    key="1"
  />
  <div
    className="starrating-container__star"
    key="2"
  />
</div>
`;
